nep-3.mediawiki

```diff
@@ -54,15 +60,18 @@ Method object has the following structure:
 {
   "name": "transfer",
   "parameters": [],
-  "returntype": "Boolean"
+  "returnType": "Boolean",
+  "readOnly": "Boolean"
 }
 </pre>
 
 <code>name</code> is the name of the method, which can be any valid identifier.
 
 <code>parameters</code> is an array of Parameter objects which describe the details of each parameter in the method.
 
-<code>returntype</code> indicates the return type of the method. It can be one of the following values: <code>Signature</code>, <code>Boolean</code>, <code>Integer</code>, <code>Hash160</code>, <code>Hash256</code>, <code>ByteArray</code>, <code>PublicKey</code>, <code>String</code>, <code>Array</code>, <code>InteropInterface</code>, <code>Void</code>.
+<code>returnType</code> indicates the return type of the method. It can be one of the following values: <code>Signature</code>, <code>Boolean</code>, <code>Integer</code>, <code>Hash160</code>, <code>Hash256</code>, <code>ByteArray</code>, <code>PublicKey</code>, <code>String</code>, <code>Array</code>, <code>InteropInterface</code>, <code>Void</code>.
+
+<code>readOnly</code> is <code>Boolean</code> which describe if the method require write access to the storage or not.
```

Comment:
I think it's not only the storage, but also the SYSCALLs like `Contract.Destroy`.